provider/pkg/provider/provider.go

```diff
@@ -683,10 +683,11 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 	}
 
 	var err error
-	namespace := "default"
-	if k.defaultNamespace != "" {
+	namespace, _, err := kubeconfig.Namespace()
+	if err != nil || k.defaultNamespace != "" {
```

Comment:
I think you want to set defaultNamespace to kubeconfig.Namespace() if it is non-empty and if the default namespace is not explicitly specified for the provider.

Reply:
Maybe I misunderstood, but `k.defaultNamespace` is the namespace configured in the `k8s.Provider`.

The original code sets `namespace` to that value, so I thought it best to do the same; I don't believe `k.defaultNamespace` is used any further?

Reply:
Though `k.defaultNamespace` is also used at line 1279. I'm not terribly familiar with this code.